content/blog/posts/vectors-matrices-matrix-order.md

```diff
@@ -0,0 +1,732 @@
+
+$$\vec{r} = \begin{bmatrix} r_{1} & r_{2} & r_{3} & ... & r_{m}  \end{bmatrix}$$
+
+Or columns:
+
+$$\vec{c} = \begin{bmatrix} c_{1} \\\ c_{2} \\\ c_{3} \\\ ... \\\ c_{m}  \end{bmatrix}$$
+
+The vector-matrix product follows the same rules as the matrix product, albeit treating row vectors as \\(1×m\\) matrices and column vectors as \\(m×1\\) matrices. Thus, for a given \\(2×2\\) matrix \\(\mathbf{M}\\) to be multiplied with a given \\(1×2\\) row vector \\(\vec{r}\\), we must ensure that the number of columns of the matrix on the left hand side matches the number of rows of the matrix on the right hand side. We achieve this by pre-multiplying the vector \\(\vec{r}\\) with \\(\mathbf{M}\\), resulting in a \\(1×2\\) matrix:
+
+$$
+\begin{array}{l@{}l}
+\vec{r} \cdot \mathbf{M} 
+    &{}= \begin{bmatrix} r_{1} & r_{2} \end{bmatrix} \cdot \begin{bmatrix} m_{11} & m_{12} \\\ m_{21} & m_{22}  \end{bmatrix} \\\ \\\
+    &{}= \begin{bmatrix} 
+r_{1}m_{11} + r_{2}m_{21} & 
+r_{1}m_{12} + r_{2}m_{22} \end{bmatrix}
+\end{array}
+$$
+
+In order to multiply a column vector by the same matrix, we must ensure that the number of columns of the matrix on the left hand side matches the number of rows of the matrix on the right hand side. We achieve this by post-multiplying the vector \\(\vec{c}\\) with \\(\mathbf{M}\\), resulting in a \\(2×1\\) matrix:
+
+$$
+\begin{array}{l@{}l}
+\mathbf{M} \cdot \vec{c} 
+    &{}= \begin{bmatrix} m_{11} & m_{12} \\\ m_{21} & m_{22}  \end{bmatrix} \cdot \begin{bmatrix} c_{1} \\\ c_{2} \end{bmatrix} \\\ \\\
+&{}= \begin{bmatrix} 
+m_{11}c_{1} + m_{21}c_{1} \\\ 
+m_{12}c_{2} + m_{22}c_{2} \end{bmatrix}
+\end{array}
+$$
+
+However, due to the non-commutativity of matrix multiplication, $\vec{r} \cdot \mathbf{M}$ may not equal to $\mathbf{M} \cdot \vec{c}$ as the two products are necessarily not equivalent:
+
+$$
+ \begin{bmatrix} 1 & 0 & 1 \end{bmatrix} \cdot \begin{bmatrix} 1 & 0 & 1 \\\ 0 & 2 & 0 \\\ 3 & 0 & 3 \end{bmatrix}     \neq \begin{bmatrix} 1 & 0 & 1 \\\ 0 & 2 & 0 \\\ 3 & 0 & 3 \end{bmatrix} \cdot \begin{bmatrix} 1 \\\ 0 \\\ 1 \end{bmatrix}
+$$
+
+ The row vector product is as follows:
+
+$$
+\begin{bmatrix} 1 & 0 & 1 \end{bmatrix} \cdot \begin{bmatrix} 1 & 0 & 1 \\\ 0 & 2 & 0 \\\ 3 & 0 & 3 \end{bmatrix} = \begin{bmatrix} 4 & 0 & 4 \end{bmatrix}
+$$
+
+Whereas the column vector product is as follows:
+
+$$
+\begin{bmatrix} 1 & 0 & 1 \\\ 0 & 2 & 0 \\\ 3 & 0 & 3 \end{bmatrix} \cdot \begin{bmatrix} 1 \\\ 0 \\\ 1 \end{bmatrix} =  \begin{bmatrix} 2 & 0 & 6 \end{bmatrix}
+$$
+
+Furthermore, the dimensions of \\(\mathbf{M}\\) may be well defined for \\(\vec{r} \cdot \mathbf{M}\\) but not necessarily for \\(\mathbf{M} \cdot \vec{c}\\). Thus, in order to make \\(\mathbf{M} \cdot \vec{c}\\) well-defined and equivalent to \\(\vec{r} \cdot \mathbf{M}\\), we must multiply \\(\vec{c}\\) by the transpose of \\(\mathbf{M}\\), such that \\(\vec{r} \cdot \mathbf{M} = \mathbf{M^T} \cdot \vec{c}\\) (and visa versa):
```

Comment:
```suggestion
Another way of thinking about this is that in order to have a well-defined product, the row vector has to appear first in the product, and since we have transposed the vector, we also need to transpose the matrix so for \\(\vec{c}\\) we would have:
```